python/ray/train/data_parallel_trainer.py

```diff
@@ -254,25 +252,69 @@ def __init__(
         train_loop_config: Optional[Dict] = None,
         backend_config: Optional[BackendConfig] = None,
         scaling_config: Optional[ScalingConfig] = None,
-        dataset_config: Optional[Dict[str, DatasetConfig]] = None,
+        dataset_config: Optional[DataConfig] = None,
         run_config: Optional[RunConfig] = None,
         datasets: Optional[Dict[str, GenDataset]] = None,
-        preprocessor: Optional["Preprocessor"] = None,
         resume_from_checkpoint: Optional[Checkpoint] = None,
+        # Deprecated.
+        preprocessor: Optional["Preprocessor"] = None,
     ):
         self._train_loop_per_worker = train_loop_per_worker
         self._train_loop_config = train_loop_config
 
+        if isinstance(dataset_config, dict) or self._dataset_config or preprocessor:
```

Comment:
should these warnings and checks be put in `BaseTrainer`? So other subclasses also have them.

Reply:
Moved the preprocessor warning.